### DISCUSSION TITLE
Dúvidas sobre API e EF

### DISCUSSION THREAD

**Discussion by JorgeSidney** [balta-io/forum#703] [Tira Dúvidas] [answered]
Pessoal, estou na parte da API das Transações e pelo que entendi, no momento temos tabelas Category e Transactions e teremos uma entidade chamada User que vai ser o assinante do software.

Pelo que entendi, no curso temos User e este User pode ter várias Transactions e as Transactions possui 1 Category permitindo consultar quantas Transactions temos por User.

No meu caso, eu gostaria que cada User pudesse ter N Clients e que cada Client possa ter N Measurements.
Aqui no curso temos o model Transaction relacionando a Category e também relacionando ao User diretamente. Se eu quero ter a tabela Measurement sendo N para cada Client, eu posso ignorar a relação de Measurement direto com o User ou devo fazer como no model abaixo?

![image](https://github.com/user-attachments/assets/57e79af2-d343-423c-abf7-aee33a9af43a)

Será que alguém pode me ajudar nesse entendimento?

Abraços.

**Comment by andrebaltieri** [comment, marked as answer]
Bom dia, @JorgeSidney, como vai?

Você quer ter um multi-tenant, ou seja, uma empresa com vários clientes tendo acesso ao sistema?

A modelagem é bem ampla, você pode tudo, e a regra é apenas a associação... Por exemplo, no caso que modelei, a categoria pertence a um usuário... ela poderia pertencer a uma empresa e todos os usuários da empresa terem acesso a ela.

Ela pode continuar assim, vinculada a um usuário, e então o usuário ser vinculado a uma empresa...

Para consultar as transações por usuário, você pode agrupar pelo UserId apenas, não precisa de uma tabela extra.

**Reply by JorgeSidney** [reply]
Fala Balta! 

Exatamente, multi-tenant. O que pensei é ter o User como se fosse o Subscriber. O Subscriber pode ter vários Clients e cada Client pode ter vários Measurements.
Então troquei de User para Subscriber, Category para Client e Transaction para Measurement, mas percebi durante o módulo de API de Transações, que eu até cheguei com essa ideia a criar o PUT e alimentar o banco corretamente, mas na hora das consultas, não veio a parte do data do json. Ou seja, tem algo errado no meu conceito de relação entre tabelas e obviamente não vai funcionar a consulta, mas fiquei com a dúvida como resolveria isso. Pensei até em contratar alguém para falar como consultoria, tentar entender um pouco mais sobre como preparar o meu Code First, até a título de curiosidade sabe? Adaptar o curso e seguir nele com o que estou prevendo.
Queria aproveitar o curso para ir montando algo mais perto da minha necessidade, ainda mais que curti o lance do curso ter formas de pagamento, algo inédito para mim.

Abraços.

Jorge Cardoso

**Reply by andrebaltieri** [reply]
Bom dia, @JorgeSidney, como vai?

Acho que podemos conversar no particular, me envia uma mensagem que te ajudo melhor:
https://balta.io/ajuda

Ai falamos mais abertamente sobre negócio e vejo como consigo te direcionar!

**Reply by JorgeSidney** [reply]
Enviei a mensagem.
Muito obrigado.

**Comment by JorgeSidney** [comment]
Obrigado.

**Comment by JorgeSidney** [comment]
Obrigado
